tmail-backend/mailets/src/main/java/com/linagora/tmail/mailet/dns/SpfDnsValidator.java

```diff
@@ -0,0 +1,213 @@
+
+    public SpfDnsValidator(DNSService dnsService, String spfAuthorizedIps) {
+        this.dnsService = dnsService;
+        this.requiredIps = parseRequiredIps(spfAuthorizedIps);
+    }
+
+    /**
+     * Validates the SPF DNS record for the given domain.
+     *
+     * @param domain the domain to validate
+     * @return Optional error message if validation fails, empty if validation succeeds
+     */
+    public Optional<String> validate(String domain) {
+        try {
+            Collection<String> txtRecords = dnsService.findTXTRecords(domain);
+
+            if (txtRecords == null || txtRecords.isEmpty()) {
+                String error = String.format("No TXT records found for domain %s", domain);
+                LOGGER.warn(error);
+                return Optional.of(error);
+            }
+
+            // Find SPF record
+            Optional<String> spfRecord = txtRecords.stream()
+                .filter(this::isSpfRecord)
+                .findFirst();
+
+            if (!spfRecord.isPresent()) {
+                String error = String.format("No SPF record found for domain %s", domain);
+                LOGGER.warn(error);
+                return Optional.of(error);
+            }
+
+            // Extract IPs from SPF record
+            Set<String> spfIps = extractIpsFromSpfRecord(spfRecord.get());
+
+            // Check if all required IPs are present
+            Set<String> missingIps = requiredIps.stream()
+                .filter(requiredIp -> !isIpAuthorized(requiredIp, spfIps))
+                .collect(Collectors.toSet());
+
+            if (!missingIps.isEmpty()) {
+                String error = String.format("SPF record for domain %s is missing required IPs: %s. Current SPF: %s",
+                    domain, String.join(", ", missingIps), spfRecord.get());
+                LOGGER.warn(error);
+                return Optional.of(error);
+            }
+
+            LOGGER.debug("SPF validation passed for domain {}", domain);
+            return Optional.empty();
+
+        } catch (Exception e) {
+            String error = String.format("Failed to query SPF record for domain %s: %s", domain, e.getMessage());
+            LOGGER.error(error, e);
+            return Optional.of(error);
+        }
+    }
+
+    @VisibleForTesting
+    Set<String> parseRequiredIps(String spfAuthorizedIps) {
+        return Splitter.on(',')
+            .trimResults()
+            .omitEmptyStrings()
+            .splitToStream(spfAuthorizedIps)
+            .collect(ImmutableSet.toImmutableSet());
+    }
+
+    @VisibleForTesting
+    boolean isSpfRecord(String txtRecord) {
+        String normalized = txtRecord.trim();
+        return normalized.startsWith(SPF_PREFIX);
+    }
+
+    @VisibleForTesting
+    Set<String> extractIpsFromSpfRecord(String spfRecord) {
+        // Extract ip4: and ip6: mechanisms from SPF record
+        // Example: "v=spf1 ip4:192.0.2.0/24 ip4:198.51.100.5 include:_spf.google.com ~all"
```

Comment:
I'm rethinking the SPF part.

It seems simpler and perfectly logic to asset that people delegate SPF management to us.

So maybe instead of doing super-complex CIDR range intersections, we could simply assert that the customer SPF policy includes ours.

The mailet would become

```
<mailet notmatch="HasSPFInclude=twake.app" class="..." \>
```

Makes sense?